Controller/SitemapController.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/*
+ * This file is part of the Symfony CMF package.
+ *
+ * (c) 2011-2014 Symfony CMF
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Cmf\Bundle\SeoBundle\Controller;
+
+use Symfony\Cmf\Bundle\SeoBundle\Model\UrlInformation;
+use Symfony\Cmf\Bundle\SeoBundle\Sitemap\UrlInformationProviderInterface;
+use Symfony\Component\Config\Definition\Exception\InvalidConfigurationException;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Templating\EngineInterface;
+
+/**
+ * Controller to handle requests for sitemap.
+ *
+ * @author Maximilian Berghoff <Maximilian.Berghoff@gmx.de>
+ */
+class SitemapController
+{
+    /**
+     * @var UrlInformationProviderInterface
+*/
+    private $chainProvider;
+    /**
+     * @var EngineInterface
+     */
+    private $templating;
+
+    /**
+     * @var array
+     */
+    private $templates;
+
+    /**
+     * Inject the templates as an hashmap like:
+     *
+     * array('format' => 'Bundle:Domain:template.format.twig')
+     *
+     * @param UrlInformationProviderInterface $provider
+     * @param EngineInterface $templating
+     * @param array $templates
+     */
+    public function __construct(
+        UrlInformationProviderInterface $provider,
+        EngineInterface $templating,
+        array $templates
+    ) {
+        $this->chainProvider = $provider;
+        $this->templating = $templating;
+        $this->templates = $templates;
+    }
+
+    /**
+     * @param string $_format The format of the sitemap. Supported values are html|xml|json
+     *
+     * @return Response
+     */
+    public function indexAction($_format)
+    {
+        $response = null;
+        $urls = $this->chainProvider->generateRoutes();
+
+        if ('json' === $_format) {
+            $response = $this->createJsonResponse($urls);
+        } elseif (isset($this->templates[$_format])) {
+            $response =  new Response($this->templating->render($this->templates[$_format], array('urls' => $urls)));
+        }
```

Comment:
maybe we could also allow other formats and remove the format requirement from the route. then you could start with isset (so that i can specify a template for json too), then try json and otherwise throw a 404 that this page is not found. with a message which formats are supported as you do below.